backend/generator.js

```diff
@@ -11,57 +11,112 @@ exports.executeScript = function (socket, formData, callback) {
     socketHandler.handleSocket(socket, 'err-logs', "Failed to generated documentation due to an error in input fields");
     return false;
   }
-  var email = formData.email;
-  var gitUrl = formData.gitUrl;
-  var docTheme = formData.docTheme;
-  var debug = formData.debug;
+  var data = [];
+  var subProjectUrl = formData.subProject === undefined ? [] : formData.subProject
   var uniqueId = uuidV4();
-  var targetBranch = formData.targetBranch === undefined ? '' : formData.targetBranch;
-  var docPath = formData.docPath === undefined ? '' : formData.docPath;
-  var subProject = "";
-  var subDocpath = [];
-  if (formData.subProject !== undefined) {
-    subProject = formData.subProject.join(",");
-    for(i=0; i<formData.subProject.length; i++){
-        subDocpath.push('docs');
-    }
-  }
-
-  const args = [
-    "-g", gitUrl,
-    "-t", docTheme,
-    "-m", email,
-    "-d", debug,
-    "-u", uniqueId,
-    "-s", subProject,
-    "-p", subDocpath.join(","),
-    "-b", targetBranch,
-    "-l", docPath
-  ];
-
-  var process = spawn("./generate.sh", args);
-
-  socketHandler.handleLineOutput(socket, process, 'logs', 4);
-  socketHandler.handleLineError(socket, process, 'err-logs');
+  var buildId = uuidV4();
+  var subProjectNames = [];
+  subProjectUrl.forEach(function (x) {
+     subProjectNames.push(x.split("/")[4].split(".")[0]);
+  });
+  var mainProjetData = {
+    email: formData.email,
+    gitUrl: formData.gitUrl,
+    docTheme: formData.docTheme,
+    debug: formData.debug,
+    uniqueId: uniqueId,
+    targetBranch: formData.targetBranch,
+    docPath: formData.docPath,
+    buildId: buildId,
+    subProjectNames: subProjectNames
+  };
 
-  process.on('exit', function (code) {
-    console.log('child process exited with code ' + code);
-    var data = { code: code, email: email, uniqueId: uniqueId, gitUrl: gitUrl };
-    if (code === 0) {
-      socketHandler.handleSocket(socket, 'success', data);
+  data.push(mainProjetData);
+  subProjectUrl.forEach(function (x) {
+    var subProjectData =  {
+      email: formData.email,
+      gitUrl: x,
+      docTheme: formData.docTheme,
+      debug: formData.debug,
+      uniqueId: uuidV4(),
+      targetBranch: formData.targetBranch,
+      docPath: "docs",
+      isSubProject: "true",
+      buildId: buildId
+    };
+    data.push(subProjectData);
+  })
+  var generatedData = Promise.all(data.map(x => generate(socket, x)))
+  generatedData.then(function (result) {
+    var flag = true;
+    var failedData;
+    result.forEach(function (x) {
+      if (x.code !== 0) {
+        flag = false;
+        failedData = x;
+      }
+    });
+    if (flag) {
```

Comment:
flag is used to check any of the project builds is failing or not. if it fails flag became false and error will be emitted in socket